numba/np/arrayobj.py

```diff
@@ -5266,3 +5264,31 @@ def impl(arr):
                        "is ambiguous. Use a.any() or a.all()")
                 raise ValueError(msg)
         return impl
+
+
+@overload(np.swapaxes)
+def numpy_swapaxes(arr, axis1, axis2):
+    if not isinstance(axis1, (int, types.Integer)):
+        raise errors.TypingError('The second argument "axis1" must be an '
+                                 'integer')
+    if not isinstance(axis2, (int, types.Integer)):
+        raise errors.TypingError('The third argument "axis2" must be an '
+                                 'integer')
+    if not isinstance(arr, types.Array):
+        raise errors.TypingError('The first argument "arr" must be an array')
+
+    # create tuple list for transpose
+    axes_list = tuple(range(arr.ndim))
+
+    def impl(arr, axis1, axis2):
+        # to ensure tuple_setitem support of negative values
+        if axis1 < 0:
+            axis1 += arr.ndim
+        if axis2 < 0:
+            axis2 += arr.ndim
+
+        axes_tuple = tuple_setitem(axes_list, axis1, axis2)
+        axes_tuple = tuple_setitem(axes_tuple, axis2, axis1)
+        return np.transpose(arr, axes_tuple)
```

Comment:
I'm reasonably convinced that this will break the "swapaxes returns a view" notion as per the docs, https://numpy.org/doc/stable/reference/generated/numpy.swapaxes.html:
> For NumPy >= 1.10.0, if a is an ndarray, then a view of a is returned; otherwise a new array is created.

Asserting that `a is expected` and `a is got` in the unittests might highlight this.

Reply:
`a is expected` and `a is got` should always be false (except when exchanging with the same axis). For being true, not only the data but also the view needs to be the same (which gets changed) or one needs to use the `base` or `np.share_memory`.

You are right, that the definition breaks the "swapaxes returns a view". But the reason is, that the numba version of `np.transpose` does not return a view. Here a short code snippet:

```python
import numpy as np
from numba import njit
b.base is a.base: True
c.base is a.base: False
@njita: False
def transpose(a):
    return np.transpose(a)
c.base is a.base: False

# reference arraype command to continue
a = np.arange(8).reshape(2, -1)

b = np.transpose(a)
c = transpose(a)

# this is always false
print(f'b is a: {b is a}')  # false
print(f'c is a: {c is a}')  # false

# using base it works
print(f'b.base is a.base: {b.base is a.base}')  # true
print(f'c.base is a.base: {c.base is a.base}')  # false
```

Reply:
I don't think there's much we can do about this right now? Do you feel that this behaviour is prohibitive to it being useful as-is?

Reply:
At least from my side there is not much I can do, I have not enough inside in the code base to propose a fix. But I think the function is useful as-is. But in general it could be helpful to document difference from numpy functions, maybe simple at
https://numba.readthedocs.io/en/stable/reference/numpysupported.html

```diff
- `np.transpose()`
+ `np.transpose()` (a copy is returned)
```

Reply:
> At least from my side there is not much I can do, I have not enough inside in the code base to propose a fix. But I think the function is useful as-is. But in general it could be helpful to document difference from numpy functions, maybe simple at
> https://numba.readthedocs.io/en/stable/reference/numpysupported.html
> 
> ```diff
> - `np.transpose()`
> + `np.transpose()` (a copy is returned)
> ```

Agree, this is probably the best thing to do for now.